comp/core/workloadmeta/dump_test.go

```diff
@@ -25,7 +25,7 @@ func TestDump(t *testing.T) {
 		fx.Supply(NewParams()),
 	))
 
-	s := newWorkloadMeta(deps).(*workloadmeta)
+	s := newWorkloadMeta(deps).Comp.(*workloadmeta)
```

Comment:
The current design is that flares should be populated by flare providers provided with their respective component. So now, the `newWorkloadmeta` function must instantiate its own flare providers at the same time as the main component. Fx will retrieve all the flare providers that were provided by all the different components and use them to generate the archive.

Basically `func newWorkloadMeta(_ deps) *workloadmeta` is converted to `func newWorkloadMeta(_ deps) (*workloadmeta, FlareProvider)`.
Fx provides a cleaner way to do it with `fx.Out`.

In practice it doesn't change anything at how we use workloadmeta because of dependency injections. During tests, because we don't have a function that initializes only the store, we must use `provider.Comp`.

Reply:
after looking the `type provider struct` I understood the respond.

but him wondering if we can do an interface `provider` with a function `GetComponent() *workoadmeta`

Reply:
For this specific use-case it might be useful but it might allow misusage of this function. IMO we can just make a function only for testing.